### DISCUSSION TITLE
Exposing defaultValues would be useful with complex controlled components

### DISCUSSION THREAD

**Discussion by aleehedl** [react-hook-form/react-hook-form#12314] [Ideas]
Hi, 

I know this has been discussed before but I haven't read a explanation why this couldn't be added. With complex controlled components, for example a rich text editor, being able to access the `defaultValues`  would be really useful. Updating the value constantly from the controller back to the component can be heavy and unnecessary since the component manages its state. Updates are only needed when the value of the `defaultValues` changes (via `reset`). 

Is there a specific reason why the value of `defaultValues` could not be exposed? Currently you can access it via `control._defaultValues` but obviously that's something you shouldn't do 👼 

Simplified example: 

```
export function MyFormContainer() {
  const formMethods = useForm({
    defaultValues: {
      name: 'foo',
      text: undefined,
    },
  });


  return (
    <FormProvider {...formMethods}>
      <MyForm />
    </FormProvider>
  )
}

export function MyForm() {
  const { control, defaultValues, register } = useFormContext();

  return (
    <form>
      <input {...register('name')} />
      <Control 
        control={control}
        name="text" 
        render={({ field: { name, onChange } }) => (
          <RichTextEditor 
            defaultValue={defaultValues.text}  // <-- This should update as seldom as possible
            name={name} 
            onChange={onChange}    
          /> 
        )} 
      />
    </form>
  )
}
```

**Comment by BrendanC23** [comment]
Is this what you're looking for?

```ts
const form = useForm({ ... });
const defaultValues = form.formState.defaultValues;
```

**Reply by aleehedl** [reply]
Yes 🙈 Thank you so much. Apparently I didn't read the docs properly and only found this outdated issue #1044.
